File: SaveTheDate.DL/Models/EventGift.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace SaveTheDate.DL.Models
{
    public partial class EventGift
    {
        public EventGift()
        {
            Guests = new HashSet<Guest>();
        }

        public int Id { get; set; }
        public int? GiftId { get; set; }
        public int EventId { get; set; }
        public int? UserId { get; set; }
        public bool? Status { get; set; }
        public string Blessing { get; set; }
        public string NewGift { get; set; }

        public virtual Event Event { get; set; }
        public virtual Gift Gift { get; set; }
        public virtual User User { get; set; }
        public virtual ICollection<Guest> Guests { get; set; }
    }
}

File: SaveTheDateAngular/src/app/Classes/EventGift.ts
import { Gift } from "./Gift";
import { User } from "./User";
import { Event } from "./Event";

  export class EventGift{
      
       Id?: number;
       GiftId!: number;
       EventId!:number;
       UserId!:number;
       Status!:boolean;
       Blessing!: string;

       Event?:Event
       Gift?:Gift
       User?:User

      constructor(){}
    };
        
